src/lib/FlightTasks/tasks/FlightTask.hpp

```diff
@@ -50,6 +50,16 @@
 
 #include "../SubscriptionArray.hpp"
 
+// this defines the output of the FlightTask library
+struct ControlSetpoint {
+	matrix::Vector3f position_setpoint;
+	matrix::Vector3f velocity_setpoint;
+	matrix::Vector3f acceleration_setpoint;
+	matrix::Vector3f thrust_setpoint;
+	float yaw_setpoint;
+	float yawspeed_setpoint;
+};
```

Comment:
I don't entirely see the benefit of a private struct. To me it seems you almost reverted to what we had in a previous flighttask version (https://github.com/PX4/Firmware/blob/cd3cf07bb6079204eab250e4ff50609fb44652ac/src/lib/FlightTasks/tasks/FlightTask.hpp#L124-L139) where the base class had setter-methods. However, we changed it because it created more overhead. 
In addition, if there are setter methods, then I don't really see why there needs to be a struct in between.